### ISSUE TITLE
Ruby LSP server crashed when using file to specify ruby version in gemfile

### ISSUE THREAD

**Issue by cyhsujohn** [Shopify/ruby-lsp#1442]
### Description

We're following [the bundler guide](https://bundler.io/guides/gemfile_ruby.html#:~:text=ruby%20file%3A%20%22.ruby%2Dversion%22) and using a file to specify the ruby version in gemfile. Ruby LSP seems to use the wrong relative location to retrieve the file or not supporting this approach.

The code we included in the gemfile: `ruby file: ".ruby-version"`

Ruby version: 3.2.3
Rails version: 7.1.3
Ruby LSP version: v0.5.11
VS code version: 1.87.0
(All updated to the latest version.)

Error message:
```
2024-03-01 17:00:09.361 [info] (easyship-api) /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler.rb:505:in `initialize':  (Bundler::Dsl::DSLError)

[!] There was an error parsing `Gemfile`: 
[!] There was an error parsing `Gemfile`: No such file or directory @ rb_sysopen - /Users/johnhsu/easyship-api/.ruby-lsp/.ruby-version. Bundler cannot continue.

 #  from /Users/johnhsu/easyship-api/Gemfile:9
 #  -------------------------------------------
 #  
 >  ruby file: ".ruby-version"
 #  
 #  -------------------------------------------
. Bundler cannot continue.

 #  from /Users/johnhsu/easyship-api/.ruby-lsp/Gemfile:4
 #  -------------------------------------------
 #  
 >  eval_gemfile(File.expand_path("../Gemfile", __dir__))
 #  gem "ruby-lsp", require: false, group: :development
 #  -------------------------------------------

	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler.rb:505:in `open'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler.rb:505:in `block in read_file'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler/shared_helpers.rb:103:in `filesystem_access'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler.rb:504:in `read_file'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler/ruby_dsl.rb:36:in `normalize_ruby_file'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler/ruby_dsl.rb:14:in `ruby'
	from /Users/johnhsu/easyship-api/Gemfile:9:in `eval_gemfile'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler/dsl.rb:49:in `instance_eval'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler/dsl.rb:49:in `eval_gemfile'
	from /Users/johnhsu/easyship-api/.ruby-lsp/Gemfile:4:in `eval_gemfile'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler/dsl.rb:49:in `instance_eval'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler/dsl.rb:49:in `eval_gemfile'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler/dsl.rb:12:in `evaluate'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler/definition.rb:37:in `build'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler.rb:208:in `definition'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler.rb:156:in `setup'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler/setup.rb:23:in `block in <top (required)>'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler/ui/shell.rb:159:in `with_level'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler/ui/shell.rb:111:in `silence'
	from /Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler/setup.rb:23:in `<top (required)>'
	from <internal:/Users/johnhsu/.rvm/rubies/ruby-3.2.3/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
	from <internal:/Users/johnhsu/.rvm/rubies/ruby-3.2.3/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
	from /Users/johnhsu/.rvm/rubies/ruby-3.2.3/lib/ruby/3.2.0/rubygems.rb:1370:in `<top (required)>'
	from <internal:gem_prelude>:2:in `require'
	from <internal:gem_prelude>:2:in `<internal:gem_prelude>'
/Users/johnhsu/.rvm/gems/ruby-3.2.3/gems/bundler-2.4.22/lib/bundler.rb:505:in `initialize':  (Bundler::Dsl::DSLError)
```

**Comment by andyw8** [comment]
Can you try with the latest Bundler? There was an [bug](https://github.com/Shopify/vscode-ruby-lsp/issues/943#issuecomment-1852902696) with older versions.

**Comment by vinistock** [comment]
Yeah, this is a bug that was fixed in Bundler v2.5.1 https://github.com/Shopify/vscode-ruby-lsp/issues/922#issuecomment-1858725522.

You can update your locked Bundler version with `bundle update --bundler`.
